MSAL/test/unit/native_auth/cache/MSALNativeAuthCacheAccessorTest.swift

```diff
@@ -27,7 +27,15 @@ import XCTest
 @_implementationOnly import MSAL_Private
 
 final class MSALNativeAuthCacheAccessorTest: XCTestCase {
-    private let cacheAccessor = MSALNativeAuthCacheAccessor()
+    private let tokenCache: MSIDDefaultTokenCacheAccessor = {
```

Comment:
Maybe add a test case on the `MSALNativeAuthPublicClientApplicationTest` class to test that the cache accessor is the same (in terms of memory pointer so same object) in the user account result as the one that was sent on SignUp.
I'm talking here about simulating a SignUp and then when the userAccount result comes back the cacheAccessor should be the same. 
It should be like the `func testSignUpPassword_correlationId_whenSetOnStart_itCascadesToAll() {` test just that at the end you could do a check like this:
`XCTAssertTrue(ObjectIdentifier(delegateSignInAfterSignUp.expectedUserAccountResult!.cacheAccessor as! MSALNativeAuthCacheAccessor) == ObjectIdentifier(cacheAccessor as! MSALNativeAuthCacheAccessor))`
Where `cacheAccessor` would be passed to the controllerFactoryMock.
Maybe there is a easier way to test this but this would cover the scenario

Reply:
As discussed there shouldn't be an End To End test to check this but rather make sure that the `cacheAccessor` uses the correct config

Reply:
I added a test for this in MSALNativeAuthPublicClientApplicationTest (along with associated factories, etc.)